lib/schemaUtils.js

```diff
@@ -198,6 +199,85 @@ module.exports = {
 
   safeSchemaFaker: safeSchemaFaker,
 
+  analyzeSpec: function (spec) {
+    var size,
+      numberOfRefs = 0,
+      specString,
+      numberOfRequests = 0;
+
+    // Stringify and add whitespaces as there would be in a normal file
+    // To get accurate disk size
+    specString = JSON.stringify(spec);
+    // Size in MB
+    size = Buffer.byteLength(specString, 'utf8') / (1024 * 1024);
+
+    // No need to check for number of requests or refs if the size is greater than 8 MB
+    // The complexity is 10.
+    if (size < 8) {
+      // Finds the number of requests that would be generated from this spec
+      if (spec.paths) {
+        Object.values(spec.paths).forEach((value) => {
+          Object.keys(value).forEach((key) => {
+            if (METHODS.includes(key)) {
+              numberOfRequests++;
+            }
+          });
+        });
+      }
+
+      // Number of times the term $ref is repeated in the spec.
+      numberOfRefs = (specString.match(/\$ref/g) || []).length;
+    }
+
+    return {
+      size,
+      numberOfRefs,
+      numberOfRequests
+    };
+  },
+
+  determineOptions: function (analysis, options) {
+    let size = analysis.size,
+      numberOfRefs = analysis.numberOfRefs,
+      numberOfRequests = analysis.numberOfRequests;
+
+    var computedOptions = _.clone(options);
+
+    computedOptions.stackLimit = 10;
+    // This is the score that is given to each spec on the basis of the
+    // number of references present in spec and the number of requests that will be generated.
+    // This ranges from 0-10.
+    computedOptions.complexityScore = 0;
+
+    // Anything above the size of 8MB will be considered a big spec and given the
+    // least stack limit and the highest complexity score.
+    if (size >= 8) {
+      console.warn('Complexity score = 10');
```

Comment:
No logs

Reply:
@abhijitkane We use console warns in other places too, right?